### PR TITLE
fix(helm): registry missing in postgresql templating

### PR DESCRIPTION
**Pull request by ion-elgreco**
## Summary & Motivation
The postgresql chart uses the "registry" value to prefix the image name. This value was not already set in the values.yaml so I had to dig into the chart.

Additionally, the Dagster helm templates don't use this registry value to create the image name. So if you do provide a custom registry: "myprivate.acr.io", then it will be correctly used in the postgresql helm, but not in the dagster initdbcontainerReady creation, since there the registry is missing, and it will fail to pull the image.

### CODE REVIEW

**Review by mlarose — APPROVED** [review]
Thanks for the contribution. Just one nit about the naming of the new var

File: helm/dagster/templates/deployment-daemon.yaml [hunk 1]
```diff
@@ -51,7 +51,7 @@ spec:
       initContainers:
         {{- if .Values.dagsterDaemon.checkDbReadyInitContainer }}
         - name: check-db-ready
-          image: {{ include "dagster.externalImage.name" $.Values.postgresql.image | quote }}
+          image: {{ include "dagster.externalPostgreImage.name" $.Values.postgresql.image | quote }}
```
Comment (ion-elgreco) [inline helm/dagster/templates/deployment-daemon.yaml:54]:
```suggestion
          image: {{ include "dagster.externalPostgresImage.name" $.Values.postgresql.image | quote }}
```

File: helm/dagster/templates/deployment-flower.yaml [hunk 2]
```diff
@@ -38,7 +38,7 @@ spec:
       initContainers:
         {{- if .Values.flower.checkDbReadyInitContainer }}
         - name: check-db-ready
-          image: "{{- $.Values.postgresql.image.repository -}}:{{- $.Values.postgresql.image.tag -}}"
+          image: {{ include "dagster.externalPostgreImage.name" $.Values.postgresql.image | quote }}
```
Comment (ion-elgreco) [inline helm/dagster/templates/deployment-flower.yaml:41]:
```suggestion
          image: {{ include "dagster.externalPostgresImage.name" $.Values.postgresql.image | quote }}
```

File: helm/dagster/templates/helpers/_deployment-webserver.tpl [hunk 3]
```diff
@@ -49,7 +49,7 @@ spec:
       initContainers:
         {{- if .Values.dagsterWebserver.checkDbReadyInitContainer }}
         - name: check-db-ready
-          image: {{ include "dagster.externalImage.name" .Values.postgresql.image | quote }}
+          image: {{ include "dagster.externalPostgreImage.name" .Values.postgresql.image | quote }}
```
Comment (ion-elgreco) [inline helm/dagster/templates/helpers/_deployment-webserver.tpl:52]:
```suggestion
          image: {{ include "dagster.externalPostgresImage.name" .Values.postgresql.image | quote }}
```

File: helm/dagster/templates/deployment-celery-queues.yaml [hunk 4]
```diff
@@ -42,7 +42,7 @@ spec:
       initContainers:
         {{- if $celeryK8sRunLauncherConfig.checkDbReadyInitContainer }}
         - name: check-db-ready
-          image: "{{- $.Values.postgresql.image.repository -}}:{{- $.Values.postgresql.image.tag -}}"
+          image: {{ include "dagster.externalPostgreImage.name" $.Values.postgresql.image | quote }}
```
Comment (ion-elgreco) [inline helm/dagster/templates/deployment-celery-queues.yaml:45]:
```suggestion
          image: {{ include "dagster.externalPostgresImage.name" $.Values.postgresql.image | quote }}
```

**Comment by ion-elgreco** [comment]
> Thanks for the contribution. Just one nit about the naming of the new var

Done!

**Pull request merged**